UI Update

andrewpking opened the pull request:
### Made significant changes to the UI

- Added in pagination
- Updated search and news page templates to match rest of theme
- Added in support for multiple ads on pages
- Footer now contains social links
- Fixed some UI bugs

andrewpking committed:
Reverse footer image.

andrewpking committed:
Nav panels now take up full width of page.

andrewpking committed:
Depracate IE styles.

andrewpking committed:
Add a class to use later for nav-sections.

andrewpking committed:
Nav sections now follow card convention from UI prototype.

andrewpking committed:
Linted front page.

andrewpking committed:
Linted and added a class to post thumbnails for style fix.

andrewpking committed:
Linted and added subclass for thumbnails.

andrewpking committed:
Resolve issue with align right in wordpress squishing images in posts.

andrewpking committed:
Merge pull request #29 from andrewpking/UI-Update

Style fixes, new card layout for nav pages, and reversed footer image. Removed IE support fully from CSS.

andrewpking committed:
Added social icons to theme.

andrewpking committed:
Footer now includes social icons.

andrewpking committed:
Removed social icons from sidebar.

andrewpking committed:
Updated styles for footer to include social icons and make them
responsive.

andrewpking committed:
Fixed a style bug on bclogo image.

andrewpking committed:
Fixed a visual bug on front-page with margin-top applied to section for
hero-widget.

andrewpking committed:
Added support for multiple ads by category. Homepage top is 'homepage'
and sidebar is 'sidebar'.

andrewpking committed:
Adjust size of homepage-ad to be narrower.

andrewpking committed:
Filled in inside of icons to have better contrast with footer image.

andrewpking committed:
Recolored to match other icon colors.

andrewpking committed:
Fixed a style issue where heading is not aligned with grid on some
pages. Fixed a semantic issue where heading is not in main on some
pages.

andrewpking committed:
Deleted pagination code from LCP, it does not support that feature.

andrewpking committed:
Upgraded pagination to live inside of core theme code for reusability on
other pages. Search now uses pagination from theme core.

andrewpking committed:
Sidebar now uses home_url for audio archive link instead of absolute
link.

andrewpking committed:
Altered footer to remove "Donate" link and changed order of "connect"
links.

andrewpking committed:
Footer H4's are now H2's to better follow page semantics.

andrewpking committed:
Reflowed hierachy to use H2 to divide up homepage and H3's for content.

andrewpking committed:
Added styles to support reflowed H2/H3's on homepage and ensure hover
elements also support focus.

andrewpking committed:
Updated search on desktop to use a button instead of an input for
submission. Added styles to support hover and focus.

andrewpking committed:
Pagination styles have higher contrast and no longer overflow on mobile.

andrewpking committed:
Fixed indentation issue on 404 page.

andrewpking committed:
Reordered staff directory and shortened names per Gregory's request.
Also hid title visually.

andrewpking committed:
Added accessibility fix to hide inactive tabs from DOM. Style fix to
improve code reuse on this page for creating arrays and rendering them.

andrewpking committed:
Fixed access issue where items that are not visible may be exposed in
DOM. Improved code reuse by adding a function for rendering show lists.
Also updated styles.

andrewpking committed:
Created a template for news page which takes advantage of pagination.

andrewpking committed:
Removed semantics that did not add any meaning to the news page.

andrewpking committed:
Merge pull request #30 from andrewpking/footer-update

UI Update - Continued

andrewpking committed:
Horizontal scroll issue fix on tablet view.

andrewpking committed:
Added a template for the listen page which displays the page content on
the bottom.

andrewpking committed:
Merge pull request #31 from andrewpking/Nav-Template-Listen

Created new template nav for listen page.

andrewpking committed:
Make sure archived shows are not displayed.

andrewpking committed:
linted this page.

andrewpking committed:
Ensure the queried post is published.

andrewpking committed:
Add edge case handling for homepage for pagination.

andrewpking committed:
Add pagination to homepage.

andrewpking committed:
Added requested banner for urgent banners.

andrewpking committed:
Merge pull request #32 from andrewpking/finishing-touches-PR3

Finishing touches pr3

andrewpking committed:
Fix PHP docs for get_pagination function to correctly assert default
value is false for is_front arg.

andrewpking committed:
For Taija to see my work.

andrewpking committed:
Now says "News Contributors" instead of "News" in tabs.

andrewpking committed:
Fixed bug in search where pagination shows up twice.

andrewpking committed:
Properly implement pagination in home.php (despite this not being used)

andrewpking committed:
Create readme

andrewpking committed:
Renamed page-news-template.php to template-news-page.php to match file
organization heuristic.

andrewpking committed:
Filled in Readme.md

andrewpking committed:
Updated links in footer for new nav structure.

andrewpking committed:
Missed a link!

andrewpking committed:
Merge pull request #33 from andrewpking/cleanup-p3

Cleanup p3

taija commented on list-category-posts/single-blog.php:
```diff
@@ -1,69 +1,59 @@
 <?php
-$lcp_display_output = '';
+$lcp_display_output = "";
```
Not worth changing, but generally single quotes are preferred to double quotes in PHP. 

Single quotes treat the content as a string, whereas double quotes allow variables to be included as well, which is slightly less effecient and should only be used where needed.

taija commented on footer.php:
```diff
@@ -3,52 +3,52 @@
 		<footer class="container" id="foot">
-			<nav aria-label="Connect with KBCS">
-
+		<h2 class="sr-only">Connect with KBCS</h2>
```
This is redundant - there shouldn't be both a heading and an aria-label. If you would like the heading to label the `<nav>`, the best way to do this would be to use an aria-labelledby instead, like so:

```html
<h2 id="menu-connect-with-kbcs" class="sr-only">Connect with KBCS</h2>
<nav aria-labelledby="menu-connect-with-kbcs" ...

taija commented on front-page.php:
```diff
@@ -62,70 +82,87 @@
-			?>
-			<article class="media"> 
-				<h2 <?php post_class() ?>><a href="<?php the_permalink(); ?>"><?php the_title();?></a></h2>
+   $sticky = get_option("sticky_posts");
```
The previous methods (using array() instead of [], using single quotes instead of single quotes, and using spaces around function parameters and other statements like `if ( $sticky )` instead of `if ($sticky)`) is perscribed by WordPress's code style guide. We try and follow this where possible. Not neccessary to change back in this case, but please avoid changing things out of this format in the future.

taija commented on functions.php:
```diff
@@ -149,6 +149,7 @@ function load_frontend_scripts() {
 			add_theme_support( 'post-thumbnails' );
 				set_post_thumbnail_size( 150, 150);
 				add_image_size( 'sidebar-ad', 370,310, true);
+				add_image_size( 'homepage-ad', 1100, 150, true);
```
Note: As you added a new image size, please remember to regenerate image thumbnails after theme release.

taija reviewed (changes requested):
This is based on code review only so far- will need to do review of the functionality as well. Note that Elizabeth and Taylor are also reviewing, so they may have addtional findings. 

Most of these are just notes, but there is one actual change needed.

TaylorYeeBC reviewed (approved):
Code reviewed and marking as approved, assuming Taija's changes are resolved.

andrewpking committed:
Update theme code to follow WordPress style guide.

andrewpking committed:
Fixed ARIA issue and update theme code to follow WordPress style guide.

andrewpking commented:
Hold up, the next button is not working properly on home page for pagination. Let me fix please!

andrewpking committed:
Moved urgent banner into load_frontend_scripts() function to resolve a
state issue.

andrewpking committed:
Resolved state issue on front page where pagination links were not
rendering the current page as non-link and defaulting to page one.

andrewpking committed:
Urgent banner no longer has pop in issues thanks to changing the way
state is handled.

andrewpking committed:
There is now animation when the notification banner loads to prevent
pop-in.

andrewpking committed:
Now using native show and hide functions in JQuery.

andrewpking commented:
### All state and UI issues introduced are resolved

- No more pop in for announcement banner when navigating between pages, now handled gracefully with animation.
- Resolved a bug with pagination on homepage where links wouldn't update properly in pagination.
- Resolved issue where a frontend script is loaded as a seperate function instead of load_frontend_scripts() in functions.php

Thanks for your patience. Sometimes I just get too excited to ship it.

<img width="1440" height="789" alt="Screenshot of announcement banner loading properly on first load in private window." src="https://github.com/user-attachments/assets/2bf5a2a9-9fe4-421c-9b96-face9dc1726f" />

taija reviewed (commented):
Code review looks good on my end; will deploy to staging and do some testing

andrewpking commented:
Found a UI bug on staff page in staging. 

![Screenshot of staff page not displaying text, only images on mobile](https://github.com/user-attachments/assets/a80fec38-de6c-458c-81d9-0bb7ce90b5c6)

Also thinking we should add code to sanitize inputs on Search queries to prevent HTML tags from being inserted.

andrewpking committed:
Override bootstrap style to prevent text content in media-body class
from being clipped on mobile.

andrewpking committed:
Added input sanitization code for search.

andrewpking commented:
Pushed fixes for both issues.

taija commented on search.php:
```diff
@@ -1,113 +1,63 @@
+        $search_query = get_search_query();
+        // sanitize search query
+        // https://developer.wordpress.org/reference/functions/sanitize_text_field/
+        $search = sanitize_text_field($search_query);
```
1. As a best practice, sanitization should be done as late as possible, so it's easy to validate that it is being done. Otherwise it's super easy to miss using it. 
2. More importantly, this isn't needed at all; get_search_query() already handles sanitization per WordPress documentation.

taija commented on search.php:
```diff
@@ -1,113 +1,63 @@
+        $search = sanitize_text_field($search_query);
+
+        $allsearch = new WP_Query(array(
+            's' => $search,
```
this should use $search_query - per previous, sanitization is already done by `get_search_query()`

taija commented on search.php:
```diff
@@ -1,113 +1,63 @@
+        ));
+        $count = $allsearch->post_count;
+        ?>
+        <span class="search-terms">"<?php echo esc_html($search_query); ?>"</span>
```
One other change request here- I'd recommend removing output of `$search_query` entirely, even with sanitization. It can be used in search manipulation attacks.

andrewpking committed:
Changes requested by Taija in search.

andrewpking committed:
Urgent banner now renders properly at up to 400% zoom.

andrewpking commented:
Found a UI bug when zoomed in on the urgent banner. Pushed a fix. ![Screenshot of text overflowing on the mobile version of the site in the urgent banner at 200 times zoom on Android](https://github.com/user-attachments/assets/f9aa8296-2871-4c1a-b0ed-aade3d0f3f9a)

andrewpking committed:
Program page now works at 400% zoom on mobile devices. Content is no
longer clipped.

andrewpking committed:
Resolved some small style issues on mobile and highly zoomed layouts.

andrewpking committed:
Resolve problem where some h1's are indented.

andrewpking commented:
Made some changes to the way content is rendered at 400% zoom on mobile. Biggest change is the programs page (which currently does not function for the schedule). Also fixed a few little CSS bugs which were annoying with headings. 

Should be all good now.

andrewpking committed:
Small content changes requested by staff.

andrewpking commented:
### Good to go

Everyone is happy with the staged site. I pushed a small content change to the footer per KBCS staff request, we **DO NOT need to stage this change**, it is a static content only change.

### Change made

 We no longer have a news department and we wanted different verbiage for newsletter. `footer.php` now reflects the static content staff requested here, including:

- Changed order of **Connect** list items
- "Newsletter" is now "Subscribe to Emails"
- **News Department** is now **Main Office**
- Updated email from news@kbcs.fm to office@kbcs.fm
- Updated phone number to main office phone tree, since news phone number does not have an assigned human

### Next steps

- Code approval from IT staff @taija @efuenzalida52 
- Schedule a time to synchronously update the KBCS website
- Potentially do a plugin audit with KBCS staff (if we have the time)

taija reviewed (approved):
Don't see any issues on my end!
